edgeless_api/src/util.rs

```diff
@@ -18,19 +26,43 @@ pub fn parse_http_host(raw: &str) -> anyhow::Result<(Proto, String, u16)> {
                 "https" => Proto::HTTPS,
                 "coap" => Proto::COAP,
                 _ => {
-                    return Err(anyhow::anyhow!("Host Parse Error"));
+                    return Err(anyhow::anyhow!("Protocol Parse Error, got '{:?}'", raw));
                 }
             };
             let port = match val[3].parse() {
                 Ok(prt) => prt,
                 Err(_) => {
-                    return Err(anyhow::anyhow!("Host Parse Error"));
+                    return Err(anyhow::anyhow!("Port Parse Error, got '{:?}'", raw));
+                }
+            };
+            let maybe_an_ip = val[2]
+                .parse::<IpAddr>() // Try to parse an IP
+                .ok() // convert to Option
+                // If None (i.e no IP), remove the eventual bracket prefix and suffix (for IPv6) then try again
+                .or_else(|| val[2].strip_prefix("[")?.strip_suffix("]")?.parse::<IpAddr>().ok());
+            let host = match maybe_an_ip {
+                Some(ip) => ip.to_string(), // WARNING the verbatim IP in `raw` may be outputed in a canonical form
+                None => {
+                    // FIXME: the val[2] could still be an hostname or dns name
```

Comment:
Not quite sure if this is actually the behavior desired for listening.

We might need two different functions (one for IPs and one for hosts), or pass the raw string through an additional IP parsing step.

In any case, the API of this is not sufficient.